### ISSUE TITLE
Updates to HOWTOs

### ISSUE THREAD

**Issue by cjfields** [bioperl/bioperl.github.io#1]
Sign up here for HOWTO doc updates.  So far:

@bosborne : Beginners, SearchIO, SeqIO,~~Restriction_Enzyme_Analysis~~
@pcantalupo : ~~BlastPlus~~, ~~Writing_BioPerl_Tests~~, ~~Getting_Genomic_Sequences~~, ~~PAML~~, ~~PhyloXML~~, ~~PopGen~~, ~~Simple_web_analysis~~, ~~SubmitPatch~~, ~~Nexml~~, ~~Glyphs~~
@majensen : ~~Tiling~~, ~~EUtilities WebServices~~, ~~Short-read_assemblies_with_BWA~~, ~~Short-read_assemblies_with_maq~~, ~~Trees~~
@cjfields : EUtilities Cookbook, ~~Feature-Annotation~~, ~~AlignIO_and_SimpleAlign~~, Bioperl_Objects, Nexml

Remaining: 
Graphics
Local_Databases
OBDA
OBDA_Flat_databases
PhylogeneticAnalysisPipeline
Trees

**Comment by majensen** [comment]
Updated list in comment-

**Comment by pcantalupo** [comment]
anybody having problems rendering an ftp link in Markdown? I cannot get

```
[blast+](ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/)
```

to show as a hyperlink.

see: [blast+](ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/)

**Comment by majensen** [comment]
hmm - is it the '+'?

**Comment by pcantalupo** [comment]
doesn't work without the plus

**Comment by majensen** [comment]
[blast+](ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/)
[blast+](http://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/)

Ah, see above, MD doesn't get "ftp" as a protocol maybe

**Comment by hyphaltip** [comment]
however [http link](http://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/) also works so you dont need the ftp protocol link.

**Comment by pcantalupo** [comment]
thank you

**Comment by pcantalupo** [comment]
Some of the HOWTOs have irritating non-breaking spaces in the code which does not look good when converted to html. I used the following command to convert them to regular spaces in BlastPlus.md:

```
perl -pe 's/\xc2\xa0/ /g' BlastPlus.md
```

Hopefully this will help somebody else.

**Comment by bosborne** [comment]
Thanks Paul, ran that over all the files.

**Comment by pcantalupo** [comment]
Is there a facility in Markdown for creating a Table of Contents like that in the HOWTO Wiki?

**Comment by bosborne** [comment]
It currently does not work:

http://ajoz.github.io/2014/06/29/i-want-my-github-flavored-markdown/

I went through the same process as this poster, trying different MD interpreters in search of one that could do code blocks. Unfortunately the one that does does not do TOC’s.

**Comment by bosborne** [comment]
Chris, I fixed the Feature-Annotation HOWTO.

**Comment by bosborne** [comment]
Keeping track, these HOWTOs look done: Beginners, Feature-Annotation, BlastPlus, Utilities_Cookbook, SearchIO, SeqIO, Getting Genomic Sequences, AlignIO and SimpleAlign, and Writing BioPerl Tests.

Remaining:

EUtilities_Web_Service
Glyphs
Graphics
Nexml
OBDA
OBDA_Flat_databases
PAML
PhyloXML
PhylogeneticAnalysisPipeline
PopGen
Restriction_Enzyme_Analysis
Short-read_assemblies_with_BWA
Short-read_assemblies_with_maq
Simple_web_analysis
SubmitPatch
Tiling
Trees
Local_Databases
Bioperl_Objects

Some of these are incomplete, never finished. Comment them out from the list online?

**Comment by bosborne** [comment]
Remaining:

EUtilities_Web_Service
Glyphs
Graphics
Nexml
OBDA
PAML
PhyloXML
PhylogeneticAnalysisPipeline
PopGen
Restriction_Enzyme_Analysis
Short-read_assemblies_with_BWA
Short-read_assemblies_with_maq
Simple_web_analysis
SubmitPatch
~~Tiling~~
Trees
Local_Databases
Bioperl_Objects

**Comment by majensen** [comment]
Tiling is merged in.

**Comment by bosborne** [comment]
Remaining:

EUtilities Web Service HOWTO
Glyphs HOWTO
Graphics HOWTO
Nexml HOWTO
PAML HOWTO
PhyloXML HOWTO
PopGen HOWTO
Restriction Enzyme Analysis HOWTO
Short-read assemblies with BWA HOWTO
Short-read assemblies with maq HOWTO
Simple web analysis HOWTO
SubmitPatch HOWTO
Trees HOWTO
Bioperl Objects HOWTO

**Comment by pcantalupo** [comment]
I'll take the P's: PAML, PhyloXML, PopGen

**Comment by majensen** [comment]
BWA is merged in.

**Comment by cjfields** [comment]
I'll work on Restriction Enzyme Analysis and Trees/PhyloXML/NeXML in addition to the Bioperl Objects I already have.  Also, thanks for the work on Feature-Annotation @bosborne !  I kinda fell off the grid for a few days taking care of a sick five-year old (who is doing a lot better now)

**Comment by bosborne** [comment]
Chris,

Restriction Enzyme Analysis is done. Yes, it’s all being taken away from you.

;-)

Here is what remains:

EUtilities Web Service HOWTO
Glyphs HOWTO
Graphics HOWTO
Nexml HOWTO
PAML HOWTO
PhyloXML HOWTO
PopGen HOWTO
Short-read assemblies with maq HOWTO
Simple web analysis HOWTO
SubmitPatch HOWTO
Trees HOWTO
Bioperl Objects HOWTO

BIO

**Comment by cjfields** [comment]
@bosborne that's completely fine by me :smile: 

I noticed some oddities with the Bioperl Objects one (some things have gone missing in the conversion).

**Comment by pcantalupo** [comment]
@cjfields I started working on PhyloXML and will be done soon. Do you want me to finish it up and commit it?

**Comment by cjfields** [comment]
Go ahead, that works for me.

On Dec 5, 2015, at 7:27 AM, Paul Cantalupo <notifications@github.com<mailto:notifications@github.com>> wrote:

@cjfieldshttps://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_cjfields&d=BQMCaQ&c=8hUWFZcy2Z-Za5rBPlktOQ&r=fbHa8Njtvh9VmSnzJxiEUTW9NWDwMMwQAzhgZDO41GQ&m=XrufMM2gcyCefwAHUIGFjhXRSMnq4jb7J_5_5rNe1HU&s=7VG0esve3tiVsbuDa3IenpZvEHFDBwmr7abwVmCoWEo&e= I started working on PhyloXML and will be done soon. Do you want me to finish it up and commit it?

**Comment by majensen** [comment]
Maq is merged in.

**Comment by majensen** [comment]
I'll do EUtil Web Svc and Trees

**Comment by cjfields** [comment]
@majensen I was planning on taking Trees but haven't started yet.  If you've started then it's all yours, just let me know.

**Comment by majensen** [comment]
Too late @cjfields , Trees merged in!

**Comment by cjfields** [comment]
All, looks like templates like `{{CPAN|Bio::Seq}}` are silently dropped; lots of these in the 'Bioperl Objects' HOWTO:  

**MW** (from the 'Brief descriptions' section')

```
===Brief descriptions===

This section describes various Bioperl sequence objects. Many people using Bioperl will never know, or need to know, what kind of sequence object they are using. This is because the {{CPAN|SeqIO}} module creates exactly the right type of object when given a file or a filehandle or a string.

{{CPAN|Bio::Seq}} is the central sequence object in bioperl. When in doubt this is probably the object that you want to use to describe a DNA, RNA or protein sequence in bioperl. Most common sequence manipulations can be performed with Seq.
```

**Converted Markdown**

```
### Brief descriptions

This section describes various Bioperl sequence objects. Many people using Bioperl will never know, or need to know, what kind of sequence object they are using. This is because the module creates exactly the right type of object when given a file or a filehandle or a string.

is the central sequence object in bioperl. When in doubt this is probably the object that you want to use to describe a DNA, RNA or protein sequence in bioperl. Most common sequence manipulations can be performed with Seq.
```

I can reconvert from the original mediawiki but it may be worth checking how prevalent these are (they mainly seemed relegated to the end).

**Comment by majensen** [comment]
They're all over the place. I have just been reading down the original howtos and adding the links in. I have a little emacs skeleton for it:

```
(define-skeleton bioperl-module-md-link
  "Insert a BioPerl module with link to MetaCPAN pod"
  "Module: "
  "[Bio::" str "](https://metacpan.org/pod/Bio::" str ")")

(global-set-key (kbd "C-c C-l") 'bioperl-module-md-link)
```

You do ctrl-c ctrl-l and then enter "SearchIO" or whatever at the 'Module:' prompt.

**Comment by majensen** [comment]
EUtilities Web Services merged in.

**Comment by pcantalupo** [comment]
@bosborne Hi, can you give us an update on what is done and what is left to do?

**Comment by bosborne** [comment]
Paul,

Looks like these are not done:

  <li><a href="/howtos/Glyphs.html">Glyphs HOWTO</a></li>
  <li><a href="/howtos/Graphics.html">Graphics HOWTO</a></li>
  <li><a href="/howtos/Nexml.html">Nexml HOWTO</a></li>
  <li><a href="/howtos/SubmitPatch.html">SubmitPatch HOWTO</a></li>

These are:

  <li><a href="/howtos/PopGen.html">PopGen HOWTO</a></li>
  <li><a href="/howtos/Beginners.html">Beginners HOWTO</a></li>
  <li><a href="/howtos/Feature-Annotation.html">Feature-Annotation HOWTO</a></li>
  <li><a href="/howtos/BlastPlus.html">BlastPlus HOWTO</a></li>
  <li><a href="/howtos/EUtilities_Cookbook.html">EUtilities Cookbook HOWTO</a></li>
  <li><a href="/howtos/SearchIO.html">SearchIO HOWTO</a></li>
  <li><a href="/howtos/SeqIO.html">SeqIO HOWTO</a></li>
  <li><a href="/howtos/Getting_Genomic_Sequences.html">Getting Genomic Sequences HOWTO</a></li>
  <li><a href="/howtos/AlignIO_and_SimpleAlign.html">AlignIO and SimpleAlign HOWTO</a></li>
  <li><a href="/howtos/Writing_BioPerl_Tests.html">Writing BioPerl Tests HOWTO</a></li>
  <li><a href="/howtos/OBDA_Flat_databases.html">OBDA Flat Databases HOWTO</a></li>
  <li><a href="/howtos/Tiling.html">Tiling HOWTO</a></li>
  <li><a href="/howtos/Local_Databases.html">Local Databases</a></li>
  <li><a href="/howtos/OBDA.html">OBDA HOWTO</a></li>
  <li><a href="/howtos/Restriction_Enzyme_Analysis.html">Restriction Enzyme Analysis HOWTO</a></li>
  <li><a href="/howtos/Short-read_assemblies_with_BWA.html">Short-read assemblies with BWA HOWTO</a></li>
  <li><a href="/howtos/Short-read_assemblies_with_maq.html">Short-read assemblies with maq HOWTO</a></li>
  <li><a href="/howtos/Trees.html">Trees HOWTO</a></li>
  <li><a href="/howtos/EUtilities_Web_Service.html">EUtilities Web Service HOWTO</a></li>
  <li><a href="/howtos/PAML.html">PAML HOWTO</a></li>
  <li><a href="/howtos/PhyloXML.html">PhyloXML HOWTO</a></li>
  <li><a href="/howtos/Bioperl_Objects.html">Bioperl Objects HOWTO</a></li>
  <li><a href="/howtos/Simple_web_analysis.html">Simple Web Analysis HOWTO</a></li>
  <li><a href="/howtos/Advanced_BioPerl.html">Advanced BioPerl HOWTO</a></li>

Brian O.

**Comment by pcantalupo** [comment]
ok, got SubmitPatch and Nexml done...I'll tackle Glyphs and Graphics next.

**Comment by pcantalupo** [comment]
@bosborne are you going to finish up Graphics HOWTO or should I take care of the rest?

**Comment by bosborne** [comment]
@pcantalupo I did a few edits on Graphics this AM, I believe it is done. And with that, we are finished this task!

**Comment by majensen** [comment]
\o/

On Thu, Dec 10, 2015 at 8:41 AM, Brian Osborne < notifications@github.com [notifications@github.com] > wrote:
Closed #1 [https://github.com/bioperl/bioperl.github.io/issues/1] .

—
Reply to this email directly or view it on GitHub
[https://github.com/bioperl/bioperl.github.io/issues/1#event-488180774] .[https://github.com/notifications/beacon/AAFAmRbBDhWpLj6vl2dKKhZjl53B8LiQks5pOXkUgaJpZM4GrmBP.gif]
